notes.md

```diff
@@ -0,0 +1,127 @@
+# aic-kube MaaS Orchestration #
+
+## Inventory/Design Schema ##
+
+Would be nice to see this get out of Git-managed flat-files and into
+an API accessible datastore that allows more fine-grained queries
+and supports inheritance/composition natively
+
+### Requirements ###
+
+1. Use profile inheritance to minimize data repitition
+2. Future-proof to support configuration items (CIs) we consider static today
+3. Support composition to allow federated source of truth
+
+### Components ###
+
+#### bootstrap_seed.yaml ####
+
+Probably needs renamed, but I see this as being the Formation data
+that specifies explicit values - hostnames, IP address assignments,
+VLAN IDs, etc...
+
+#### bootstrap_hwdefinition.yaml ####
+
+Looks to be mainly focused on initial OOB control and getting a server
+into PXE mode. Should be able to leave as-is and adopt existing
+Apollo logic for IPMI.
+
+#### bootstrap_profiles.yaml ####
+
+Currently doesn't exist, but I think it makes sense to separate the
+data that describes the design (like general profiles) from the data
+that ties the design to explicit assignments (i.e. Formation data)
+
+### Questions/Issues ###
+
+* Need DNS specs in YAML - where to include it, possibly per network
+making it effective when that network is chosen as "primary"
+	* Can set comma-delimited list of DNS servers per subnet
+* Consider YAML pointers and anchors to diminish how much inheritance
+has to be calculated in code
+	* Does not appear to support key subtraction
+	* No way to merge hashes based on a 'primary' key
+	* Scott thinks best to keep with managing inheritance in code. Can minimize
```

Comment:
I dig a person who can refer to themselves in the third person.  I agree with all your points -- code seems best. I just thought the YAML pointers was worth a peek.